src/Action/SetObjectFieldValueAction.php

```diff
@@ -115,7 +115,20 @@ public function __invoke(Request $request): JsonResponse
 
         // Handle date and datetime types have setter expecting a DateTime object
         if ('' !== $value && \in_array($fieldDescription->getType(), ['date', 'datetime'], true)) {
-            $value = new \DateTime($value);
+            $inputTimezone = new \DateTimeZone(date_default_timezone_get());
```

Comment:
For the current underlying timezone, I think we could simply use `null`.
[`DateTime::__construct()`](https://www.php.net/manual/en/datetime.construct.php#refsect1-datetime.construct-parameters):
>timezone
>A DateTimeZone object representing the timezone of $time.
>
>If $timezone is omitted, the current timezone will be used.

Reply:
But we do not need a date in the current time zone. We need to convert the date in the user time zone to the system (default) time zone.

Reply:
From the documentation perspective, "current timezone" is actually what `date_default_timezone_get()` returns.